drivers/storage/digitalocean/tests/digitalocean_test.go

```diff
@@ -0,0 +1,365 @@
+	vols, err := client.API().Volumes(nil, 0)
+	assert.NoError(t, err)
+	if err != nil {
+		t.FailNow()
+	}
+	// Filter volumes to those under the digitalocean service,
+	// and find a volume matching inputted volume ID
+	assert.Contains(t, vols, digitalocean.Name)
+	for _, vol := range vols[digitalocean.Name] {
+		if vol.ID == volumeID {
+			return vol
+		}
+	}
+	// No matching volumes found
+	t.FailNow()
+	t.Error("failed volumeByID")
+	return nil
+}
+
+func volumeRemove(t *testing.T, client types.Client, volumeID string) {
+	log.WithField("volumeID", volumeID).Info("removing volume")
+	err := client.API().VolumeRemove(nil, digitalocean.Name, volumeID)
+
+	assert.NoError(t, err)
+	if err != nil {
+		t.Error("failed volumeRemove")
+		t.FailNow()
+	}
+}
+
+func volumeAttach(
+	t *testing.T, client types.Client, volumeID string) *types.Volume {
+	log.WithField("volumeID", volumeID).Info("attaching volume")
+
+	reply, token, err := client.API().VolumeAttach(
+		nil, digitalocean.Name, volumeID, &types.VolumeAttachRequest{})
+
+	assert.NoError(t, err)
+	if err != nil {
+		t.Error("failed volumeAttach")
+		t.FailNow()
+	}
+	apitests.LogAsJSON(reply, t)
+	assert.NotEqual(t, token, "")
+
+	return reply
+}
+
+//func volumeInspect(
+//	t *testing.T, client types.Client, volumeID string) *types.Volume {
+//	log.WithField("volumeID", volumeID).Info("inspecting volume")
+//	reply, err := client.API().VolumeInspect(nil, digitalocean.Name, volumeID, 0)
+//	assert.NoError(t, err)
+//
+//	if err != nil {
+//		t.Error("failed volumeInspect")
+//		t.FailNow()
+//	}
+//	apitests.LogAsJSON(reply, t)
+//	return reply
+//}
+
+func volumeInspectAttached(
+	t *testing.T, client types.Client, volumeID string) *types.Volume {
+	log.WithField("volumeID", volumeID).Info("inspecting volume")
+
+	reply, err := client.API().VolumeInspect(
+		nil, digitalocean.Name, volumeID,
+		types.VolAttReqTrue)
+	assert.NoError(t, err)
+
+	if err != nil {
+		t.Error("failed volumeInspectAttached")
+		t.FailNow()
+	}
+	apitests.LogAsJSON(reply, t)
+	assert.Len(t, reply.Attachments, 1)
+	return reply
+}
+
+func volumeInspectDetached(
+	t *testing.T, client types.Client, volumeID string) *types.Volume {
+	log.WithField("volumeID", volumeID).Info("inspecting volume")
+
+	reply, err := client.API().VolumeInspect(
+		nil, digitalocean.Name, volumeID,
+		types.VolAttReqTrue)
+
+	assert.NoError(t, err)
+	if err != nil {
+		t.Error("failed volumeInspectDetached")
+		t.FailNow()
+	}
+
+	apitests.LogAsJSON(reply, t)
+	assert.Len(t, reply.Attachments, 0)
+	return reply
+}
+
+func volumeInspectDetachedFail(
+	t *testing.T, client types.Client, volumeID string) *types.Volume {
+
+	log.WithField("volumeID", volumeID).Info("inspecting volume")
+	reply, err := client.API().VolumeInspect(nil, digitalocean.Name, volumeID, 0)
+	assert.NoError(t, err)
+
+	if err != nil {
+		t.Error("failed volumeInspectDetachedFail")
+		t.FailNow()
+	}
+	apitests.LogAsJSON(reply, t)
+	assert.Len(t, reply.Attachments, 1)
```

Comment:
This currently fails since `VolumeInspect` is not returning attachments if the volume isn't attached anywhere. See the comment I left in the `VolumeInspect` implementation for details.